File: mobile/app/models/buy_request.ts
import {Event} from './event'
import {UserPublic} from './user_public'


export class BuyRequest {
    
    public id:number;
    public event:Event;
    public submitted_by_user:UserPublic;
    public price_exact:number;
    public loc_long:number;
    public loc_lat:number;
    public loc_range:number;
    public state:string;
    
    constructor() {}
    
    public static fromJSON(jsonObj): BuyRequest {
        var buy_request = new BuyRequest();
        buy_request.id = jsonObj.id;
        buy_request.event = Event.fromJSON(jsonObj.event);
        buy_request.submitted_by_user = UserPublic.fromJSON(jsonObj.submitted_by_user);
        buy_request.price_exact = jsonObj.price_exact;
        buy_request.loc_long = jsonObj.loc_long;
        buy_request.loc_lat = jsonObj.loc_lat;
        buy_request.state = jsonObj.state;
        return buy_request;
    }
}

File: api/models/buy_request.py
from app import db
from event import Event
from user import User


class BuyRequest(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    cre_dt = db.Column(db.DateTime)
    del_dt = db.Column(db.DateTime)
    event_id = db.Column(db.Integer, db.ForeignKey(Event.id))
    event = db.relationship(Event, uselist=False)
    submitted_by_user_id = db.Column(db.Integer, db.ForeignKey(User.id))
    submitted_by_user = db.relationship(User, uselist=False)
    price_exact = db.Column(db.Integer)
    loc_long = db.Column(db.Float)
    loc_lat = db.Column(db.Float)
    loc_range = db.Column(db.Float)
    state = db.Column(db.String(80))

    def __init__(self):
        self.event = None
        self.price_exact = None
        self.loc_long = None
        self.loc_lat = None
        self.loc_range = None
        self.state = None
